### PR TITLE
v6.3: Add description about SM3 encryption authentication

### PR DESCRIPTION
**Pull request by CbcWestwolf**
### First-time contributors' checklist 

- [x] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### Which TiDB version(s) do your changes apply to? (Required)

Add description about SM3 encryption authentication

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [x] v6.3 (TiDB 6.3 versions)
- [ ] v6.2 (TiDB 6.2 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s): https://github.com/pingcap/tidb/pull/36193

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: security-compatibility-with-mysql.md [hunk 1]
```diff
@@ -26,6 +26,7 @@ TiDB 目前支持的身份验证方式可在以下的表格中查找到。服务
 | `sha256_password`        | 否               |
 | `caching_sha2_password`  | 是（5.2.0 版本起） |
 | `auth_socket`            | 是（5.3.0 版本起） |
+| `tidb_sm3_password`      | 是（6.3.0 版本起） |
 | [TLS Certificates]       | 是              |
```
Comment (ran-huang) [inline security-compatibility-with-mysql.md:30]:
```suggestion
| [TLS 证书]       | 是              |
```

**Review by ran-huang — APPROVED** [review]
LGTM

File: basic-features.md [hunk 2]
```diff
@@ -134,17 +134,18 @@ aliases: ['/docs-cn/dev/basic-features/']
...
-| [静态加密 (TDE)](/encryption-at-rest.md)                           |  Y  |  Y  |  Y  |  Y  |  Y  |  Y  |  Y  |  Y  |  Y  |
-| [基于角色的访问控制 (RBAC)](/role-based-access-control.md)         |  Y  |  Y  |  Y  |  Y  |  Y  |  Y  |  Y  |  Y  |  Y  |
-| [证书鉴权](/certificate-authentication.md)                         |  Y  |  Y  |  Y  |  Y  |  Y  |  Y  |  Y  |  Y  |  Y  |
-| `caching_sha2_password` 认证                                       |  Y  |  Y  |  Y  |  Y  |  Y  |  Y  |  N  |  N  |  N  |
-| [与 MySQL 兼容的 `GRANT` 权限管理](/privilege-management.md)       |  Y  |  Y  |  Y  |  Y  |  Y  |  Y  |  Y  |  Y  |  Y  |
-| [动态权限](/privilege-management.md#动态权限)                      |  Y  |  Y  |  Y  |  Y  |  Y  |  Y  |  Y  |  N  |  N  |
-| [安全增强模式](/system-variables.md#tidb_enable_enhanced_security) |  Y  |  Y  |  Y  |  Y  |  Y  |  Y  |  Y  |  N  |  N  |
-| [日志脱敏](/log-redaction.md)                                     |  Y  |  Y  |  Y  |  Y  |  Y  |  Y  |  Y  |  Y  |  N  |
+| 安全                                                            | 6.3 | 6.2 | 6.1 | 6.0 | 5.4 | 5.3 | 5.2 | 5.1 | 5.0 | 4.0 |
```
Comment (ran-huang) [inline basic-features.md:137]: basic feature is updated in this PR. FYI @shichun-0415

File: basic-features.md [hunk 3]
```diff
@@ -134,17 +134,18 @@ aliases: ['/docs-cn/dev/basic-features/']
...
+| ------------------------------------------------------------------ | :-: | :-: | :-: | :-: | :-: | :-: | :-: | :-: | :-: | :-: |
+| [传输层加密 (TLS)](/enable-tls-between-clients-and-servers.md)    |  Y  |  Y  |  Y  |  Y  |  Y  |  Y  |  Y  |  Y  |  Y  |  Y  |
+| [静态加密 (TDE)](/encryption-at-rest.md)                          |  Y  |  Y  |  Y  |  Y  |  Y  |  Y  |  Y  |  Y  |  Y  |  Y  |
+| [基于角色的访问控制 (RBAC)](/role-based-access-control.md)        |  Y  |  Y  |  Y  |  Y  |  Y  |  Y  |  Y  |  Y  |  Y  |  Y  |
+| [证书鉴权](/certificate-authentication.md)                        |  Y  |  Y  |  Y  |  Y  |  Y  |  Y  |  Y  |  Y  |  Y  |  Y  |
+| `caching_sha2_password` 认证                                      |  Y  |  Y  |  Y  |  Y  |  Y  |  Y  |  Y  |  N  |  N  |  N  |
+| `tidb_sm3_password` 认证                                      |  Y  |  N  |  N  |  N  |  N  |  N  |  N  |  N  |  N  |  N  |
```
Comment (Oreoxmt) [inline basic-features.md:144]: 这两个认证要加个链接吗，看起来只能链到 `default_authentication_plugin` 系统变量

**Review by Oreoxmt — APPROVED** [review]
Rest LGTM

File: basic-features.md (hunk 3 above)
Comment (CbcWestwolf) [reply basic-features.md]: Done.

**Review by Oreoxmt — APPROVED** [review]
LGTM

File: basic-features.md [hunk 4]
```diff
@@ -134,17 +134,18 @@ aliases: ['/docs-cn/dev/basic-features/']
...
-| [日志脱敏](/log-redaction.md)                                     |  Y  |  Y  |  Y  |  Y  |  Y  |  Y  |  Y  |  Y  |  N  |
+| 安全                                                            | 6.3 | 6.2 | 6.1 | 6.0 | 5.4 | 5.3 | 5.2 | 5.1 | 5.0 | 4.0 |
+| ------------------------------------------------------------------ | :-: | :-: | :-: | :-: | :-: | :-: | :-: | :-: | :-: | :-: |
+| [传输层加密 (TLS)](/enable-tls-between-clients-and-servers.md)    |  Y  |  Y  |  Y  |  Y  |  Y  |  Y  |  Y  |  Y  |  Y  |  Y  |
+| [静态加密 (TDE)](/encryption-at-rest.md)                          |  Y  |  Y  |  Y  |  Y  |  Y  |  Y  |  Y  |  Y  |  Y  |  Y  |
+| [基于角色的访问控制 (RBAC)](/role-based-access-control.md)        |  Y  |  Y  |  Y  |  Y  |  Y  |  Y  |  Y  |  Y  |  Y  |  Y  |
+| [证书鉴权](/certificate-authentication.md)                        |  Y  |  Y  |  Y  |  Y  |  Y  |  Y  |  Y  |  Y  |  Y  |  Y  |
+| [`caching_sha2_password` 认证](/system-variables.md#defaultauthenticationplugin)    |  Y  |  Y  |  Y  |  Y  |  Y  |  Y  |  Y  |  N  |  N  |  N  |
+| [`tidb_sm3_password` 认证](/system-variables.md#defaultauthenticationplugin)        |  Y  |  N  |  N  |  N  |  N  |  N  |  N  |  N  |  N  |  N  |
```
Comment (Oreoxmt) [inline basic-features.md:144]:
```suggestion
| [`caching_sha2_password` 认证](/system-variables.md#default_authentication_plugin)    |  Y  |  Y  |  Y  |  Y  |  Y  |  Y  |  Y  |  N  |  N  |  N  |
| [`tidb_sm3_password` 认证](/system-variables.md#default_authentication_plugin)        |  Y  |  N  |  N  |  N  |  N  |  N  |  N  |  N  |  N  |  N  |
```

**Review by Oreoxmt — APPROVED** [review]

**Comment by ran-huang** [comment]
@ljun0712 PTAL. Thanks.

File: security-compatibility-with-mysql.md [hunk 5]
```diff
@@ -26,7 +26,8 @@ TiDB 目前支持的身份验证方式可在以下的表格中查找到。服务
 | `sha256_password`        | 否               |
 | `caching_sha2_password`  | 是（5.2.0 版本起） |
 | `auth_socket`            | 是（5.3.0 版本起） |
-| [TLS Certificates]       | 是              |
+| `tidb_sm3_password`      | 是（6.3.0 版本起） |
+| [TLS 证书]       | 是              |
```
Comment (Oreoxmt) [inline security-compatibility-with-mysql.md:30]:
```suggestion
| TLS 证书       | 是              |
```

**Review by Oreoxmt — APPROVED** [review]
Rest LGTM

**Review by ljun0712 — APPROVED** [review]

**Pull request merged**